### DISCUSSION TITLE
clinicalEnrichment & plotEnrichmentResults - Mixed Reference Groups

### DISCUSSION THREAD

**Discussion by HHayman** [PoisonAlien/maftools#713] [Q&A]
Heylo!

This seemed more like a question, rather than an issue as such, but let me know if it would have been better in 'Issues'.

When using clinicalEnrichment and plotEnrichmentResults, we obtain a plot as below;

![Example](https://user-images.githubusercontent.com/75019173/118653205-3170d400-b7df-11eb-8612-565195b5ec09.png)

We see the '0' group used as group 1, then suddenly in 9th gene swaps and uses '1' as group 2. The relationships is the same, '0' having a lesser % mutated. As far as I can tell, the p value, which is used to rank the genes, is the same whichever way round you have the groups.

I'm wondering, how is it then decided which way it is plotted? For example, we get the same information (I think) if the 9th gene used '0' group as group 1, and this would be easier to interpret visually as we are not flipping the groups around. 

Would it be clearer to interpret if we uniformly used '0' or '1' as group 1? This would also make it easier to instantly see which genes are relatively enriched in a positive or negative manner. Additionally, if the 'rest' bar was also coloured by its group, this might also make it clearer visually?

Is there a purpose behind the switch? I think it might be quite jarring for an audience who will instinctively be looking for a reason why the groups are switched.

I fear I may be missing something obvious :)

Thanks!

**Comment by PoisonAlien** [comment]
Hi,

Here the function is doing two things. 

1. Groupwise: Mutations enriched in each level v/s rest of the samples 
2. Pairwise: Mutations enriched in each level v/s individual levels

Plotting is done only for GroupWise comparison. For your data since you have only two levels (0 and 1), both comparisons will probably yield the same results. 

For plotting, I can see how it can be confusing since the bars are plotting without any order. For now, a quick fix would be sort the results by `Group1` column and it should be much cleaner. 

From the vignette example:

```r
fab.ce = clinicalEnrichment(maf = laml, clinicalFeature = 'FAB_classification')
plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05)
``` 

![image](https://user-images.githubusercontent.com/8164062/118767929-a7784800-b87e-11eb-925e-e66547734dc0.png)


Sort the results by Group1 and p-value (also keep only significant results based on OR and p-value:
```r
fab.ce$groupwise_comparision = fab.ce$groupwise_comparision[order(Group1, p_value)][OR > 1 & p_value < 0.05]
plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05)
```

![image](https://user-images.githubusercontent.com/8164062/118769099-16a26c00-b880-11eb-8450-f02e2675b620.png)


Let me know if this helps.

**Comment by HHayman** [comment]
That definitely helps. As far as I can tell, we can't choose what is plotted (plotEnrichmentResults) by OR, only by p_value, is that right? I did make some attempts and checked the code, doesn't look like it. 

Arranging the results by group is definitely interesting. Is there a way to cut a group from the plot, so only those with group 1 as reference are plotted?

I guess the problem is that we can customise the results output, but not so much plotEnrichmentResults?

Thanks :)

**Reply by PoisonAlien** [reply]
I have addressed this issue in #715 
You should be able to fine control the plotting.
